In [208]:
#Import Dependencies
import pandas as pd

In [209]:
# The path to our CSV file
file = "Resources/purchase_data.csv"

# Read our purchase data into pandas
df = pd.read_csv(file)
df.head()

Purchase ID             SN  Age Gender  Item ID  \
0            0        Lisim78   20   Male      108   
1            1    Lisovynya38   40   Male      143   
2            2     Ithergue48   24   Male       92   
3            3  Chamassasya86   24   Male      100   
4            4      Iskosia90   23   Male      131   

                                   Item Name  Price  
0  Extraction, Quickblade Of Trembling Hands   3.53  
1                          Frenzied Scimitar   1.56  
2                               Final Critic   4.88  
3                                Blindscythe   3.27  
4                                       Fury   1.44


### Player Count

* Total Number of Players

In [210]:
# Total Number of Players
unique_count = df["SN"].unique()
df["SN"].value_counts()
player_count = len(df["SN"].value_counts())
player_count_disp = pd.DataFrame({"Player Count": [player_count]})
player_count_disp

Player Count
0           576

### Purchasing Analysis (Total)

* Number of Unique Items
* Average Purchase Price
* Total Number of Purchases
* Total Revenue

In [211]:
# Number of Unique Items
df["Item ID"].unique()
df["Item ID"].value_counts()
Number_uniqueItem = len(df["Item ID"].value_counts())
Number_uniqueItem 
 

183

In [212]:
#Average Purchase Price
AvgPrice = df["Price"].mean()
AvgPrice


3.050987179487176

In [213]:
#Total Number of Purchases
Total_purchase = len(df["Item Name"])
Total_purchase

780

In [214]:
#Total Revenue
Revenue = df["Price"].sum()
Revenue

2379.77

In [215]:
# Create new DataFrame to make summary table
Purchasing_Analysis_Total = pd.DataFrame({"Number of Unique Items": [Number_uniqueItem],
                                           "Average Price": [AvgPrice],
                                           "Number of Purchases": [Total_purchase],
                                           "Total Revenue": [Revenue]})

# DataFrame formatting
Purchasing_Analysis_Total["Average Price"] = Purchasing_Analysis_Total["Average Price"].map("${:.2f}".format)
Purchasing_Analysis_Total["Total Revenue"] = Purchasing_Analysis_Total["Total Revenue"].map("${:.2f}".format)
Purchasing_Analysis_Total = Purchasing_Analysis_Total[["Number of Unique Items", "Average Price", "Number of Purchases", "Total Revenue"]]

Purchasing_Analysis_Total

Number of Unique Items Average Price  Number of Purchases Total Revenue
0                     183         $3.05                  780      $2379.77

### Gender Demographics

* Percentage and Count of Male Players
* Percentage and Count of Female Players
* Percentage and Count of Other / Non-Disclosed

In [216]:
#Count of Male,Female and other/non-disclosed Players
gender = df[["SN", "Gender"]]

gender = gender.drop_duplicates()

gender_count = gender["Gender"].value_counts()

gender_count

Male                     484
Female                    81
Other / Non-Disclosed     11
Name: Gender, dtype: int64

In [217]:
#Percent of Male,Female and other/non-disclosed Players
gender_Perc = (gender_count/player_count)*100
gender_Perc = gender_Perc .round(2)
gender_Perc


Male                     84.03
Female                   14.06
Other / Non-Disclosed     1.91
Name: Gender, dtype: float64

In [218]:
gender_demographics = pd.DataFrame({"Total Count":gender_count, "Percentage of players": gender_Perc})

gender_demographics["Percentage of players"]=gender_demographics["Percentage of players"].map("{:}%".format)
gender_demographics

Total Count Percentage of players
Male                           484                84.03%
Female                          81                14.06%
Other / Non-Disclosed           11                 1.91%

### Purchasing Analysis (Gender)

* The below each broken by gender
  * Purchase Count
  * Average Purchase Price
  * Total Purchase Value
  * Average Purchase Total per Person by Gender

In [219]:
gender = df.groupby(["Gender"])
purchase_count = gender["SN"].count()
avg_purch_price = gender["Price"].mean()
purchase_value = gender["Price"].sum()
# Average Purchase Total per Person by Gender
#duplicates = df.drop_duplicates(subset='SN', keep="first")
#grouped_dup = duplicates.groupby(["Gender"])
avg_total = purchase_value / gender_count 
# Convert to DataFrame
PurchAnalys_gend = pd.DataFrame({"Purchase Count": purchase_count, "Average Purchase Price": avg_purch_price, "Total Purchase Value": purchase_value , "Avg PurchaseTotal per Person ":avg_total})
PurchAnalys_gend["Average Purchase Price"] = PurchAnalys_gend["Average Purchase Price"].map("${:,.2f}".format)
PurchAnalys_gend["Total Purchase Value"] = PurchAnalys_gend["Total Purchase Value"].map("${:,.2f}".format)
PurchAnalys_gend["Avg PurchaseTotal per Person "] = PurchAnalys_gend["Avg PurchaseTotal per Person "].map("${:,.2f}".format)
PurchAnalys_gend

Purchase Count Average Purchase Price  \
Gender                                                         
Female                            113                  $3.20   
Male                              652                  $3.02   
Other / Non-Disclosed              15                  $3.35   

                      Total Purchase Value Avg PurchaseTotal per Person   
Gender                                                                    
Female                             $361.94                         $4.47  
Male                             $1,967.64                         $4.07  
Other / Non-Disclosed               $50.19                         $4.56

### Age Demographics

* The below each broken into bins of 4 years (i.e. &lt;10, 10-14, 15-19, etc.)
  * Purchase Count
  * Average Purchase Price
  * Total Purchase Value
  * Average Purchase Total per Person by Age Group


In [220]:
bins = [0,9,14,19,24,29,34,39,60]
group_names = ["<10", "10-14", "15-19", "20-24", "25-29", "30-34", "35-39", "40+"]
df["Age group"] = pd.cut(df["Age"], bins, labels=group_names)
# Create new data frame with the added "Age Group" 
age_grouped = df.groupby("Age group")
# Count total players by age category
age_count = age_grouped["SN"].nunique()
# Calculate percentages by age category 
age_Perc = (age_count/player_count)*100
age_Perc = age_Perc .round(2)

# Create data frame with obtained values
age_demographics = pd.DataFrame({"Total Count":age_count, "Percentage of players": age_Perc})
age_demographics["Percentage of players"]= age_demographics["Percentage of players"].map("{:,.2f}%".format)
age_demographics

Total Count Percentage of players
Age group                                   
<10                 17                 2.95%
10-14               22                 3.82%
15-19              107                18.58%
20-24              258                44.79%
25-29               77                13.37%
30-34               52                 9.03%
35-39               31                 5.38%
40+                 12                 2.08%

In [221]:
#Purchase Count
purch_count_age = age_grouped["SN"].count()
#Average Purchase Price
Avgprice_age = age_grouped["Price"].mean()
#Total Purchase Value
purchase_value = age_grouped["Price"].sum()
#Average Purchase Total per Person by Age Group
avg_total_age = purchase_value/age_count
PurchAnalys_age = pd.DataFrame({"Purchase Count": purch_count_age, "Average Purchase Price": Avgprice_age, "Total Purchase Value":purchase_value , "Avg PurchaseTotal per Person ":avg_total_age})
PurchAnalys_age["Average Purchase Price"] = PurchAnalys_age["Average Purchase Price"].map("${:,.2f}".format)
PurchAnalys_age["Total Purchase Value"] = PurchAnalys_age["Total Purchase Value"].map("${:,.2f}".format)
PurchAnalys_age["Avg PurchaseTotal per Person "] = PurchAnalys_age["Avg PurchaseTotal per Person "].map("${:,.2f}".format)
PurchAnalys_age

Purchase Count Average Purchase Price Total Purchase Value  \
Age group                                                               
<10                    23                  $3.35               $77.13   
10-14                  28                  $2.96               $82.78   
15-19                 136                  $3.04              $412.89   
20-24                 365                  $3.05            $1,114.06   
25-29                 101                  $2.90              $293.00   
30-34                  73                  $2.93              $214.00   
35-39                  41                  $3.60              $147.67   
40+                    13                  $2.94               $38.24   

          Avg PurchaseTotal per Person   
Age group                                
<10                               $4.54  
10-14                             $3.76  
15-19                             $3.86  
20-24                             $4.32  
25-29                             $3.81  
30-34                             $4.12  
35-39                             $4.76  
40+                               $3.19

### Top Spenders

* Identify the the top 5 spenders in the game by total purchase value, then list (in a table):
  * SN
  * Purchase Count
  * Average Purchase Price
  * Total Purchase Value

In [271]:
spender =df.groupby(["SN"])
#Purchase Count
purch_count_spender = spender["Purchase ID"].count()
#Average Purchase Price
Avgprice_spender = spender["Price"].mean()
#Total Purchase Value
totalpurchase_spender = spender["Price"].sum()

Topspenders = pd.DataFrame({"Purchase Count": purch_count_spender, "Average Purchase Price": Avgprice_spender, "Total Purchase Value":totalpurchase_spender })
Topspenders["Average Purchase Price"] = Topspenders["Average Purchase Price"].map("${:,.2f}".format)
Topspenders["Total Purchase Value"] = Topspenders["Total Purchase Value"].map("${:,.2f}".format)

#Sort in descending order to obtain top 5 spender names 
formatted_spenders = Topspenders.sort_values("Purchase Count", ascending=False)
formatted_spenders.head()

Purchase Count Average Purchase Price Total Purchase Value
SN                                                                     
Lisosia93                 5                  $3.79               $18.96
Iral74                    4                  $3.40               $13.62
Idastidru52               4                  $3.86               $15.45
Asur53                    3                  $2.48                $7.44
Inguron55                 3                  $3.70               $11.11

### Most Popular Items

* Identify the 5 most popular items by purchase count, then list (in a table):
  * Item ID
  * Item Name
  * Purchase Count
  * Item Price
  * Total Purchase Value

In [268]:
item_data = df.loc[:,["Item ID", "Item Name", "Price"]]
items = item_data.groupby(["Item ID","Item Name"])
#Purchase Count
purch_count_item = items["Price"].count()
#Total Purchase Value
Purchtotal_item = items["Price"].sum()
#Item Price
item_price = Purchtotal_item/purch_count_item 
popularItem = pd.DataFrame({"Purchase Count": purch_count_item,"Item Price":item_price, "Total Purchase Value":Purchtotal_item })
popularItem["Total Purchase Value"] = popularItem["Total Purchase Value"].map("${:,.2f}".format)
popularItem["Item Price"] = popularItem["Item Price"].map("${:,.2f}".format)
popularItem = popularItem.sort_values("Purchase Count", ascending=False)
popularItem.head()

Purchase Count  \
Item ID Item Name                                                      
178     Oathbreaker, Last Hope of the Breaking Storm              12   
145     Fiery Glass Crusader                                       9   
108     Extraction, Quickblade Of Trembling Hands                  9   
82      Nirvana                                                    9   
19      Pursuit, Cudgel of Necromancy                              8   

                                                     Item Price  \
Item ID Item Name                                                 
178     Oathbreaker, Last Hope of the Breaking Storm      $4.23   
145     Fiery Glass Crusader                              $4.58   
108     Extraction, Quickblade Of Trembling Hands         $3.53   
82      Nirvana                                           $4.90   
19      Pursuit, Cudgel of Necromancy                     $1.02   

                                                     Total Purchase Value  
Item ID Item Name                                                          
178     Oathbreaker, Last Hope of the Breaking Storm               $50.76  
145     Fiery Glass Crusader                                       $41.22  
108     Extraction, Quickblade Of Trembling Hands                  $31.77  
82      Nirvana                                                    $44.10  
19      Pursuit, Cudgel of Necromancy                               $8.16

### Most Profitable Items

* Identify the 5 most profitable items by total purchase value, then list (in a table):
  * Item ID
  * Item Name
  * Purchase Count
  * Item Price
  * Total Purchase Value


In [272]:
item_data = df.loc[:,["Item ID", "Item Name", "Price"]]
items = item_data.groupby(["Item ID","Item Name"])
#Purchase Count
purch_count_item = items["Price"].count()
#Total Purchase Value
Purchtotal_item = items["Price"].sum()
#Item Price
item_price = Purchtotal_item/purch_count_item 
popularItem2 = pd.DataFrame({"Purchase Count": purch_count_item,"Item Price":item_price, "Total Purchase Value":Purchtotal_item })
popularItem2["Total Purchase Value"] = popularItem2["Total Purchase Value"].map("${:,.2f}".format)
popularItem2["Item Price"] = popularItem2["Item Price"].map("${:,.2f}".format)
popularItem2 = popularItem2.sort_values("Total Purchase Value", ascending=False)
popularItem.head()

Purchase Count  \
Item ID Item Name                                                      
178     Oathbreaker, Last Hope of the Breaking Storm              12   
145     Fiery Glass Crusader                                       9   
108     Extraction, Quickblade Of Trembling Hands                  9   
82      Nirvana                                                    9   
19      Pursuit, Cudgel of Necromancy                              8   

                                                     Item Price  \
Item ID Item Name                                                 
178     Oathbreaker, Last Hope of the Breaking Storm      $4.23   
145     Fiery Glass Crusader                              $4.58   
108     Extraction, Quickblade Of Trembling Hands         $3.53   
82      Nirvana                                           $4.90   
19      Pursuit, Cudgel of Necromancy                     $1.02   

                                                     Total Purchase Value  
Item ID Item Name                                                          
178     Oathbreaker, Last Hope of the Breaking Storm               $50.76  
145     Fiery Glass Crusader                                       $41.22  
108     Extraction, Quickblade Of Trembling Hands                  $31.77  
82      Nirvana                                                    $44.10  
19      Pursuit, Cudgel of Necromancy                               $8.16

OBSERVATIONS:
*From gender wise analysis, it is obvious that male players are 84% where as women are only 14%.So male spend more time on this than women.
* From age wise analysis,we can conclude that most of the players are from age group 15-30.In it 20-24 age group has a majority with 45%.
* 20-24 age group is the ones who spend more money to purchase too.From purchasing analysis(Age),it is very clear.
* Oathbreaker, Last Hope of the Breaking Storm is the most popular game
